en/security.rst

```diff
@@ -18,8 +18,298 @@ ACL (Access Control List)
 Authorization
 =============
 
+.. _tde:
 
 TDE (Transparent Data Encryption)
 =================================
 
+.. _tde-overview:
 
+CUBRID TDE Concept
+------------------
+
+CUBRID supports **Transparent Data Encryption (henceforth, TDE)**. TDE means transparently encrypting data from the user's point of view. This allows users to encrypt data stored on disk with little to no application change.
+
+CUBRID TDE provides encryption and decryption at the engine level to minimize performance degradation due to encryption. When a user creates an encrypted table, all relevant user data stored on disk (data at rest) is automatically encrypted. By providing TDE, CUBRID helps users to comply with security regulations and guidelines required in various sites.
+
+**Table Encryption**
+
+In CUBRID, a **table** is the unit for TDE-encryption. To use the TDE feature, create a table using the **ENCRYPT** option as follows. For more information, see :ref:`create-tde-table`.
+
+.. code-block:: sql
+
+	CREATE TABLE tde_tbl (att1 INT, att2 VARCHAR(20)) ENCRYPT=AES;
+
+When an encrypted table is created, all data related to the table is automatically encrypted when written to disk, and decrypted when read into memory. Related data includes not only tables, but also indexes created on the table, temporary data created while executing queries related to the table, logs created when data is changed, DWB, and backups. For more details, see :ref:`tde-enc-target` and :ref:`tde-restriction`.
+
+.. _tde-key:
+
+Key Management
+--------------
+
+CUBRID uses symmetric key algorithms to encrypt the data. Keys used for encryption are managed in two levels consisting of master keys and data keys for efficiency. Master keys managed by the user is stored in a separate file, and CUBRID provides a utility to manage it.
+
+.. _tde-2level-key:
+
+2-Level Key Management
+^^^^^^^^^^^^^^^^^^^^^^
+
+CUBRID TDE manages keys in two levels as follows:
+
+.. image:: /images/tde_2_level_key.png
+  :width: 469
+  :align: center
+  :alt: 2 Level Key Management Image
+
+*    **Master key**: A key used when encrypting and decrypting data keys, and is managed by DBA user.
+*    **Data Key**: A key used when encrypting user data such as table and log, and is managed by CUBRID Engine.
+
+Data keys are stored within the data volume and are always securely encrypted using a master key when written to disk. The master key is stored in a separate file and it must be managed safely according to a security policy users comply with.
+
+Managing keys in two levels makes it possible to efficiently perform the key change operation. If there is only a key that encrypts the user data, when you change the key, it takes a long time to work because all the data which has been encrypted has to be read, and decrypted and re-encrypted. Also, the overall performance of the database may be degraded during this process.
```

Comment:
```suggestion
Managing keys in two levels make it possible to perform the key change operation efficiently. If there is only a key that encrypts the user data, it takes a long time to work when you change the key. All the data that has been encrypted has to be read, decrypted, and re-encrypted. Also, the overall performance of the database may be degraded during this process.
```

- Comment: Sentence might be unclear or hard to follow, to make it short, I rephrased it.

Reply:
the verb 'makes' shoulds stay as 'makes', my mistake.